htdocs/core/triggers/interface_20_all_Logevents.class.php

```diff
@@ -76,103 +80,70 @@ public function runTrigger($action, $object, User $user, Translate $langs, Conf
 			$conf->entity = $entity; // forcing of the entity if it's not defined (ex: in login form)
 		}
 
-		$date = dol_now();
-
-		/* Actions */
-
-		$text = '';
-		$desc = '';
-		$langs->load("users");
-
 		// Actions
 		switch ($action) {
 			case 'USER_LOGIN':
 				dol_syslog("Trigger '" . $this->name . "' for action '$action' launched by " . __FILE__ . ". id=" . $object->id);
-				// Initialize data (date,duree,text,desc)
-				$text = "(UserLogged," . $object->login . ")";
-				$desc = "(UserLogged," . $object->login . ")";
+				$this->initEventData("UserLogged", $object, $langs, true);
 				break;
 			case 'USER_LOGIN_FAILED':
 				dol_syslog("Trigger '" . $this->name . "' for action '$action' launched by " . __FILE__ . ". id=" . $object->id);
+				$this->initEventData("UserLoginFailed", $object, $langs, true);
 				break;
 			case 'USER_LOGOUT':
 				dol_syslog("Trigger '" . $this->name . "' for action '$action' launched by " . __FILE__ . ". id=" . $object->id);
-				// Initialize data (date,duree,text,desc)
-				$text = "(UserLogoff," . $object->login . ")";
-				$desc = "(UserLogoff," . $object->login . ")";
+				$this->initEventData("UserLogoff", $object, $langs, true);
 				break;
 			case 'USER_CREATE':
 				dol_syslog("Trigger '" . $this->name . "' for action '$action' launched by " . __FILE__ . ". id=" . $object->id);
-				// Initialize data (date,duree,text,desc)
-				$text = $langs->transnoentities("NewUserCreated", $object->login);
-				$desc = $langs->transnoentities("NewUserCreated", $object->login);
+				$this->initEventData("NewUserCreated", $object, $langs, true);
 				break;
 			case 'USER_MODIFY':
 				dol_syslog("Trigger '" . $this->name . "' for action '$action' launched by " . __FILE__ . ". id=" . $object->id);
-				// Initialize data (date,duree,text,desc)
-				$text = $langs->transnoentities("EventUserModified", $object->login);
-				$desc = $langs->transnoentities("EventUserModified", $object->login);
+				$this->initEventData("EventUserModified", $object, $langs, true);
 				break;
 			case 'USER_NEW_PASSWORD':
 				dol_syslog("Trigger '" . $this->name . "' for action '$action' launched by " . __FILE__ . ". id=" . $object->id);
-				// Initialize data (date,duree,text,desc)
-				$text = $langs->transnoentities("NewUserPassword", $object->login);
-				$desc = $langs->transnoentities("NewUserPassword", $object->login);
+				$this->initEventData("NewUserPassword", $object, $langs, true);
 				break;
-			case 'USER_ENABLEDISABLE':
+			case 'USER_ENABLEDISABLE': // TODO maybe divide this action into 2 separate actions for better traceability
```

Comment:
We try to do the opposite:  reduce the number of trigger codes (in perfect word, we should have only trigger CREATE, UPDATE, DELETE and use the property of object to make the difference between actions.

Reply:
Ok, thanks for explaining this. I think this will need more substantial work and should be done separately.